File: src/main/java/cz/vhromada/catalog/mapper/PictureMapper.java
package cz.vhromada.catalog.mapper;

import cz.vhromada.catalog.entity.Picture;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for picture.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface PictureMapper {

    /**
     * Maps entity picture to domain picture.
     *
     * @param source entity picture
     * @return mapped domain picture
     */
    cz.vhromada.catalog.domain.Picture map(Picture source);

    /**
     * Maps domain picture to entity picture.
     *
     * @param source domain picture
     * @return mapped entity picture
     */
    Picture mapBack(cz.vhromada.catalog.domain.Picture source);

}

File: src/main/kotlin/cz/vhromada/catalog/mapper/PictureMapper.kt
package cz.vhromada.catalog.mapper

import cz.vhromada.catalog.entity.Picture
import cz.vhromada.common.mapper.Mapper
import org.springframework.stereotype.Component

/**
 * A class represents mapper for picture.
 *
 * @author Vladimir Hromada
 */
@Component("pictureMapper")
class PictureMapper : Mapper<Picture, cz.vhromada.catalog.domain.Picture> {

    override fun map(source: Picture): cz.vhromada.catalog.domain.Picture {
        return cz.vhromada.catalog.domain.Picture(
                id = source.id,
                content = source.content!!.copyOf(),
                position = source.position)
    }

    override fun mapBack(source: cz.vhromada.catalog.domain.Picture): Picture {
        return Picture(
                id = source.id,
                content = source.content.copyOf(),
                position = source.position)
    }

}
